### PR TITLE
doc: update documentation of persist

### PR DESCRIPTION
**Pull request by wcastand**
## Related Issues

Fixes #1510 

## Summary

Update doc to include how to handle Map and Set with the new persist API.

issue since i have yarn v3 installed, it creates `.yarn` and `.yarnrc.yml` and need me to set pacakge to v1 with `yarn set version classic` so not sure i can run prettier easily but i will run it and delete the files we don't want commited.

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Review by dai-shi — COMMENTED** [review]
Thanks for the PR. LGTM

File: docs/integrations/persisting-store-data.md [hunk 1]
```diff
@@ -571,3 +571,34 @@ export const useBearStore = create<MyState>()(
   )
 )
 ```
+
+### How do I use deal with Map/Set with the persist api
+
+With the previous persist API, you would use `serialize`/`deserialize` to deal with Map and Set and convert them into Array so they could be parsed into proper JSON.
```
Comment (chrisk-7777) [inline docs/integrations/persisting-store-data.md:577]:
```suggestion
With the previous persist API, you would use `serialize`/`deserialize` to deal with Map and Set and convert them into an array so they could be parsed into proper JSON.
```

Comment (chrisk-7777) [inline docs/integrations/persisting-store-data.md:577]:
```suggestion
With the previous persist API, you would use `serialize`/`deserialize` to deal with Map and Set and convert them into an Array so they could be parsed into proper JSON.
```

File: docs/integrations/persisting-store-data.md [hunk 2]
```diff
@@ -571,3 +571,34 @@ export const useBearStore = create<MyState>()(
...
+The new persist API has deprecated `serialize`/`deserialize`.
+
+Now, you will need to use the `storage` prop, let's say your state use a `Map` to handle a list of `transactions`, then you can convert the Map into an Array in the storage prop:
```
Comment (chrisk-7777) [inline docs/integrations/persisting-store-data.md:580]:
```suggestion
Now, you will need to use the `storage` prop. Let's say your state uses `Map` to handle a list of `transactions`, then you can convert the Map into an Array in the storage prop:
```

**Review by chrisk-7777 — CHANGES_REQUESTED** [review]
Looks good, I just have two really nit-picky change requests

File: docs/integrations/persisting-store-data.md [hunk 3]
```diff
@@ -571,3 +571,34 @@ export const useBearStore = create<MyState>()(
...
+
+```typecript
```
Comment (sewera) [inline docs/integrations/persisting-store-data.md:582]: There is a typo in `typecript`. Please fix it either to `typescript`, or better yet, simply use `ts`.

File: docs/integrations/persisting-store-data.md [hunk 4]
```diff
@@ -571,3 +571,34 @@ export const useBearStore = create<MyState>()(
   )
 )
 ```
+
+### How do I use deal with Map/Set with the persist api
```
Comment (sewera) [inline docs/integrations/persisting-store-data.md:575]: I'm not completely sure how to understand it, but I think it will be more readable with a shorter question:

```suggestion
### How do I use it with Map and Set?
```

We implicitly use the `persist` API on this doc page, so I omitted it.

File: docs/integrations/persisting-store-data.md (hunk 1 above)
Comment (sewera) [inline docs/integrations/persisting-store-data.md:577]: If I understood it correctly, this is how I'd rephrase it.

```suggestion
With the previous persist API, you would use `serialize`/`deserialize`
to deal with mapping and setting the data and converting it
into an Array, so it could be parsed into a proper JSON.
```

File: docs/integrations/persisting-store-data.md (hunk 2 above)
Comment (sewera) [inline docs/integrations/persisting-store-data.md:580]:
```suggestion
Now, you will need to use the `storage` prop.
For example, let's say your state uses `Map` to handle
a list of transactions, then you can convert that Map
into an Array in the `storage` prop:
```

**Review by sewera — CHANGES_REQUESTED** [review]

File: docs/integrations/persisting-store-data.md (hunk 1 above)
Comment (wcastand) [reply docs/integrations/persisting-store-data.md:577]: Not really, i'm talking about the JS Object `Map` and `Set` not mapping and setting.

Comment (sewera) [reply docs/integrations/persisting-store-data.md]: Alright, you can wrap it in backticks, so `Map` and `Set`, so that it's clear that they're JS objects.

Comment (sewera) [reply docs/integrations/persisting-store-data.md]: And you can split this long line to shorter ones, to match the rest of this file.

**Comment by sewera** [comment]
The PR looks good to me, nice work :)

@chrisk-7777, I think you can re-review and approve those changes now.

**Comment by dai-shi** [comment]
I can merge an approval from either of you.

**Review by chrisk-7777 — APPROVED** [review]
thank you for actioning those changes - look good now 👍
